File: Models/Finance/OperatorPayout.cs
using TicketPortal.Api.Models.Common;
using TicketPortal.Api.Models.CompanyNetwork;
using TicketPortal.Api.Models.Enums;
using System;
using System.ComponentModel.DataAnnotations;

namespace TicketPortal.Api.Models.Finance
{
    // A record of money WE paid OUT to the operator (the "PlatformPaysOperator" direction —
    // e.g. sending them their share of a week's online sales, minus our commission).
    public class OperatorPayout : AuditableEntity
    {
        public Guid BusOperatorId { get; set; }
        public Guid? OperatorSettlementId { get; set; } // Which settlement run this payout is for.

        [MaxLength(50)]
        public string PayoutNo { get; set; } = string.Empty;

        public decimal Amount { get; set; }

        [MaxLength(3)]
        public string Currency { get; set; } = "BDT";

        public PayoutStatus Status { get; set; } = PayoutStatus.Pending;
        public DateTime? PaidAtUtc { get; set; }

        [MaxLength(120)]
        public string? BankTransactionReference { get; set; }

        [MaxLength(250)]
        public string? Notes { get; set; }

        public BusOperator BusOperator { get; set; } = default!;
        public OperatorSettlement? OperatorSettlement { get; set; }
    }
}

File: apps/admin/src/console/types/operatorPayout.ts
// Mirrors OperatorPayoutResponseDto / CreateDto / CompleteDto / ActionDto.
export type PayoutStatus = "Pending" | "Processing" | "Paid" | "Failed" | "Cancelled";

export interface OperatorPayout {
  id: string;
  busOperatorId: string;
  operatorSettlementId: string | null;
  payoutNo: string;
  amount: number;
  currency: string;
  status: PayoutStatus;
  paidAtUtc: string | null;
  bankTransactionReference: string | null;
  notes: string | null;
  createdAtUtc: string;
  updatedAtUtc: string | null;
  rowVersion: string;
}

export interface OperatorPayoutCreatePayload {
  busOperatorId: string;
  operatorSettlementId?: string | null;
  amount: number;
  currency: string;
  notes?: string | null;
}

export interface OperatorPayoutCompletePayload {
  bankTransactionReference: string;
}

export interface OperatorPayoutActionPayload {
  reason: string;
}
